GradientDescent.ipynb

```diff
@@ -42,18 +49,37 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 2,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Not implemented.\n"
+     ]
+    },
+    {
+     "ename": "AttributeError",
+     "evalue": "'NotImplementedType' object has no attribute 'reshape'",
+     "output_type": "error",
+     "traceback": [
+      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
+      "\u001B[0;31mAttributeError\u001B[0m                            Traceback (most recent call last)",
+      "\u001B[0;32m<ipython-input-2-01c779630ff9>\u001B[0m in \u001B[0;36m<module>\u001B[0;34m\u001B[0m\n\u001B[1;32m      8\u001B[0m \u001B[0mtheta_pred\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mNotImplemented\u001B[0m \u001B[0;31m# TODO: Implement the analytical solution\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m      9\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m---> 10\u001B[0;31m \u001B[0mprint\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'Empirical theta'\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mtheta_pred\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mreshape\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m-\u001B[0m\u001B[0;36m1\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m     11\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n",
+      "\u001B[0;31mAttributeError\u001B[0m: 'NotImplementedType' object has no attribute 'reshape'"
+     ]
+    }
+   ],
    "source": [
     "print('Not implemented.')\n",
     "\n",
     "'''\n",
     "Hints:\n",
-    "1. Use np.matmul() and la.inv() to solve for x in Ax = b.\n",
+    "1. Use np.matmul() and la.pinv() to solve for x in Ax = b.\n",
```

Comment:
Maybe la.pinv() takes away the work they need to do vs. writing the pseudoinverse out with A.T and A? Could ask the professor what she thinks, ultimately up to you if you want to keep it